tests/unittests/Foundation/NSThreadTests.m

```diff
@@ -77,6 +78,24 @@ - (void)countingloop {
     testController.stop = YES;
 }
 
+TEST(NSThread, ExternalCurrentThreadLeak) {
+    NSThread* current = nil;
+
+    std::thread t([&current]() {
+        // Grab a strong reference to NSThread.currentThread
+        // on a thread not created via the NSThread APIs
+        current = [[NSThread currentThread] retain];
+    });
+
+    t.join();
+
+    // Ensure the reference grabbed above is
+    // the only one remaining after thread exit
+    EXPECT_EQ(1, [current retainCount]);
```

Comment:
I am not sure I understand how this is correct.  alloc init should add a reference and currentThread retain should add one?